frontend/src/component/application/ConnectedInstances/ConnectedInstances.tsx

```diff
@@ -1,99 +1,38 @@
+import { useApplicationOverview } from '../../../hooks/api/getters/useApplicationOverview/useApplicationOverview';
+import { useConnectedInstances } from '../../../hooks/api/getters/useConnectedInstances/useConnectedInstances';
 
-const Container = styled('div')(({ theme }) => ({
-    '* + *': {
-        marginBlockStart: theme.spacing(2),
-    },
-}));
-
-const EnvironmentSelectionContainer = styled('div')(({ theme }) => ({
-    label: {
-        '--padding-horizontal': theme.spacing(3),
-        '--padding-vertical': theme.spacing(1),
-        color: theme.palette.primary.main,
-        background: theme.palette.background,
-        paddingInline: 'var(--padding-horizontal)',
-        paddingBlock: 'var(--padding-vertical)',
-        border: `1px solid ${theme.palette.background.alternative}`,
-        borderInlineStart: 'none',
-        fontWeight: 'bold',
-        position: 'relative',
-
-        svg: {
-            color: theme.palette.warning.main,
-            position: 'absolute',
-            fontSize: theme.fontSizes.bodySize,
-            top: 'calc(var(--padding-horizontal) * .12)',
-            right: 'calc(var(--padding-horizontal) * .2)',
-        },
-    },
-    'label:first-of-type': {
-        borderInlineStart: `1px solid ${theme.palette.background.alternative}`,
-        borderRadius: `${theme.shape.borderRadiusMedium}px 0 0 ${theme.shape.borderRadiusMedium}px`,
-    },
-    'label:last-of-type': {
-        borderRadius: `0 ${theme.shape.borderRadiusMedium}px ${theme.shape.borderRadiusMedium}px 0`,
-    },
-    'label:has(input:checked)': {
-        background: theme.palette.background.alternative,
-        color: theme.palette.primary.contrastText,
-
-        svg: {
-            color: 'inherit',
-        },
-    },
-    'label:focus-within': {
-        outline: `2px solid ${theme.palette.background.alternative}`,
-        outlineOffset: theme.spacing(0.5),
-    },
-
-    fieldset: {
-        border: 'none',
-        padding: 0,
-        margin: 0,
-    },
-    legend: {
-        marginBlockEnd: theme.spacing(3),
-    },
-
-    '.visually-hidden': {
-        border: 0,
-        clip: 'rect(0 0 0 0)',
-        height: 'auto',
-        margin: 0,
-        overflow: 'hidden',
-        padding: 0,
-        position: 'absolute',
-        width: '1px',
-        whiteSpace: 'nowrap',
-    },
-}));
-
-export const ConnectedInstances = () => {
+export const ConnectedInstances: FC = () => {
     const name = useRequiredPathParam('name');
     const { application } = useApplication(name);
-    const [currentEnvironment, setCurrentEnvironment] =
-        useQueryParam('environment');
-    const availableEnvironments = new Set(
-        application?.instances.map(
-            // @ts-expect-error: the type definition here is incomplete. It
-            // should be updated as part of this project.
-            (instance) => instance.environment,
-        ),
+    const { data: applicationOverview } = useApplicationOverview(name);
```

Comment:
once I add tests I will be adding a custom hook to simplify the low level logic here